horace_core/sqw/file_io/@binfile_v4_common/put_senpix_block.m

```diff
@@ -0,0 +1,61 @@
+function  obj = put_senpix_block(obj,img_block,pos)
+%PUT_SENPIX_BLOCK stores whole input image data or part of image data
+% replacing all or part of current image data.
```

Comment:
Let's try again. img_block contains s,e,npix. The basic task of this is to write it to the file described by obj. It is overwriting part of the image in the file. The start of the image modifications is given by pos. The end of the image modifications is given by pos+size of (relevant bit of) img_block. Neither of these endpoints may fall outside the location of the existing image in the file. If they equal the start and end of the image on file, then the whole of that image is modified. Otherwise only part of the image is modified. Modification of the image is limited to a single stretch of data; writing of particular cells in the image is not covered. The whole of the image in img_block (not part of it) is written

If this is the case, please put some form of this in as a description of the method. At the moment the major omissions are "current image data ON FILE" and "whole or part -> single subsection of the image data". Also the statement "stores whole input image data or part of it" suggests that not all of img_block will be written whereas I think the whole thing is written overwriting part of the image on file.

If this is not the case, then provide an equivalently detailed description of what is actually happening

Reply:
this is exactly what it is and you apparenlty understood this. So what is the problem?

Reply:
Apparntly you want the whole image format descrition provided in this one routine. I can do it now but it is not how usually methods are described. You can not describe all file format in each method of the class.

Reply:
@abuts 

Alex

The whole point about my review comment is that your opening comment for the function does NOT say what I said. 

As per my comment

At the moment the major omissions are "current image data ON FILE" and "whole or part -> single subsection of the image data". Also the statement "stores whole input image data or part of it" suggests that not all of img_block will be written whereas I think the whole thing is written overwriting part of the image on file.

These are what needs documenting. I do not need a full description of the image format.

I will additionally note that this would be a lot simpler if this function only replaced a complete image. Replacing part of the image in this way begs the question of why only this partial replacement is supported while other partial replacements are not.

I am not approving this until this is addressed